backend/db/deck.go

```diff
@@ -0,0 +1,232 @@
+package db
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/joaquinleonarg/wdml_mtg/backend/domain"
+	"github.com/rs/zerolog/log"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func GetDeckById(id string) (*domain.Deck, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	dbDeckId, err := primitive.ObjectIDFromHex(id)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrInvalidID, err)
+	}
+	// Find pack
+	result := MongoDatabaseClient.
+		Database(DB_MAIN).
+		Collection(COLLECTION_DECKS).
+		FindOne(ctx,
+			bson.M{"_id": dbDeckId},
+		)
+	if err := result.Err(); err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, fmt.Errorf("%w: %v", ErrNotFound, err)
+		}
+		return nil, fmt.Errorf("%w: %v", ErrInternal, err)
+	}
+
+	// Decode deck
+	var deck *domain.Deck
+	err = result.Decode(&deck)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrInternal, err)
+	}
+	return deck, nil
+}
+
+func GetDecksByTournamentPlayerId(id string) ([]domain.Deck, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	dbTournamentPlayerID, err := primitive.ObjectIDFromHex(id)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrInvalidID, err)
+	}
+	// Find pack
+	filter := bson.D{{Key: "tournament_player_id", Value: dbTournamentPlayerID}}
+	cursor, err := MongoDatabaseClient.
+		Database(DB_MAIN).
+		Collection(COLLECTION_DECKS).
+		Find(ctx, filter)
+	if err != nil {
+		if err == mongo.ErrNoDocuments {
+			return nil, fmt.Errorf("%w: %v", ErrNotFound, err)
+		}
+		return nil, fmt.Errorf("%w: %v", ErrInternal, err)
+	}
+
+	// Decode deck
+	var decks []domain.Deck
+	err = cursor.All(ctx, &decks)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrInternal, err)
+	}
+	return decks, nil
+}
+
+func CreateEmptyDeck(deck domain.Deck) error {
+	if deck.ID != primitive.NilObjectID {
+		return ErrObjectIDProvided
+	}
+	deck.ID = primitive.NewObjectID()
+	deck.Cards = make([]domain.DeckCard, 0)
+	deck.CreatedAt = primitive.NewDateTimeFromTime(time.Now())
+	deck.UpdatedAt = primitive.NewDateTimeFromTime(time.Now())
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	// Begin transaction
+	session, err := MongoDatabaseClient.
+		StartSession()
+	if err != nil {
+		return fmt.Errorf("%w: %v", ErrInternal, err)
+	}
+	defer session.EndSession(ctx)
+
+	_, err = session.WithTransaction(ctx, func(ctx mongo.SessionContext) (interface{}, error) {
+		resultInsert, err := MongoDatabaseClient.
+			Database(DB_MAIN).
+			Collection(COLLECTION_DECKS).
+			InsertOne(ctx,
+				deck,
+			)
+		if err != nil {
+			return nil, fmt.Errorf("%w: %v", ErrInternal, err)
+		}
+		return resultInsert, nil
+	})
+
+	log.Debug().Str("created deck id", deck.ID.String())
```

Comment:
```suggestion
	log.Debug().Str("deck_id", deck.ID.String()).Msg("created deck")
```
Igual despues deberíamos mejorar el debugging en db en general

Reply:
Creo que te falto esto

Reply:
yep, me lo comi